Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
+
+            if (cancellationToken.IsCancellationRequested
+                || notification.Reaction.User.Value.IsBot
+                || notification.Message.Value.Embeds.Count < 1)
+                return;
+
+            if (_memoryCache.TryGetValue(notification.Message.Id, out CachedPromoPoll poll)
+            && ValidateVoteReaction(reaction))
+            {
+                await notification.Message.Value.RemoveReactionAsync(notification.Reaction.Emote, notification.Reaction.User.Value);
+
+                var sentiment = reaction.Name == _approveEmoji.Name ? PromotionSentiment.Approve : PromotionSentiment.Oppose;
+
+                if (!await _promotionsService.VotePromoAsync(poll.Campaign.Id, user,  sentiment))
+                    return;
+
+                if (reaction.Name == _approveEmoji.Name)
+                    ++poll.Approve;
+                else if(reaction.Name == _disproveEmoji.Name)
+                    ++poll.Disprove;
+
+                await poll.Message.ModifyAsync(m => m.Embed = BuildPollEmbed(poll.Campaign, poll.Approve, poll.Disprove).Build());
+            }
+        }
+
+        private async Task CreatePromoPoll(PromotionActionCreationData campaign)
+        {
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+            var message = await pollChannel.SendMessageAsync(embed: BuildPollEmbed(campaign.Campaign, 1, 0).Build());
+
+            await message.AddReactionsAsync(new[] {(IEmote)_approveEmoji, _disproveEmoji});
+
+            var poll = new CachedPromoPoll
+            {
+                Message = message,
+                Campaign = campaign.Campaign,
+                Approve = 1,
+                Disprove = 0
+            };
+
+            _memoryCache.Set(campaign.Campaign.Id, poll);
+            _memoryCache.Set(message.Id, poll);
+        }
+
+        private async Task DeletePromoPoll(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(campaign.Campaign.Id, out CachedPromoPoll poll))
+                return;
+            await poll.Message.DeleteAsync();
+        }
+
+        private EmbedBuilder BuildPollEmbed(PromotionCampaignBrief campaign, int Approve, int Disprove)
+        {
+                var boldRole = $"**{MentionUtils.MentionRole(campaign.TargetRole.Id)}**";
+                var user = _discordSocketClient.GetUser(campaign.Subject.Id);
+
+                ValueTask<Color> aviColor;
+
+                if ((user.GetAvatarUrl(size: 16) ?? user.GetDefaultAvatarUrl()) is { } avatarUrl)
+                {
+                    aviColor = _imageService.GetDominantColorAsync(new Uri(avatarUrl));
+                }
+                else
+                {
+                    aviColor = new ValueTask<Color>(Color.Teal);
+                }
+
+                return new EmbedBuilder()
```

Comment:
Add a hyperlink to the full voting page on the web UI. E.G. "Click here if you'd like to leave a comment with your vote".